### ISSUE TITLE
Feature: Secure Self Service reset

### ISSUE THREAD

**Issue by johanpeters** [akeeba/loginguard#64]
it would be nice to have a Secure Self Service reset availible for users.
i don't think all users will nicely have the backup codes printed.. 
so if a user is out of 2 factor options te get in.. a Sercure Self Service reset would be nice.. 
with the result that after reset, on next login the user will start configure the 2 factor again.

It has to be secure.. so .. lets see.. secret question and answer for example.
Self configured at the beginning of 2 factor configuration.. 
so the user needs to know current password, what the secret question and answer ..
if al those are correctly answered the users 2 factor configuration wil be reset, 
a password change and a notify to the admin

i am not a programmer so i don't know if its possible / complex to build.. 
but for maintaining a website with 100+ users.. you don't whan't to get mails and calls every day of resetting there 2 factor ... and basicly.. i don't know everyone.. so i don't know 100% that its the correct user..

**Comment by nikosdion** [comment]
:man_facepalming:

**Comment by nikosdion** [comment]
Two Step Verification is meant to protect your account against credentials compromise and brute forcing. What you are proposing is that we implement a procedure to disable Two Step Verification which can be brute forced or subverted by an attacker. If you do not see a problem with that you should stop offering Two Step Verification to your clients. It's far more honest than offering "security" which is broken by design (therefore **INSECURE**, the polar opposite of security!).

FYI, here are all the secure ways to work around TSV:

1. Backup codes. This is exactly what they are for.
1. Multiple 2SV methods. For example, having Google Authenticator and a security key.
1. In case of hardware authenticators (YubiKey, security keys) have two of them, one locked in a safe.
1. If you are the owner of the site and you failed at sensible security, i.e. did not do any of the first three things, you can disable the plugin by renaming it, regain access to your site and reset the 2SV.
1. If you are anyone else, contact the Super User of the site to unblock you. The Super User must verify your identity (e.g. ask for a scan of the passport with all but the full name, nationality and last four digits of the passport number blacked out). Then and only then will they unblock you.

Yes, the latter is a hassle. I have thousands of active users on my site and I get 1-2 emails about blocked 2SV every week. Big fucking deal! The fact that my clients are protected against account subversion if far more important than spending 20' every week. If you don't want to be bothered with that just don't offer 2SV.
